### PR TITLE
Sequencer L1 transacting improvements

### PR DESCRIPTION
**Pull request by PlasmaPower**
- Uses the new validator replace by fee method, which with #1444 includes full EIP-1559 support
- Guarantees that on startup or on failover, the sequencer sequences delayed messages before incoming user transactions (note: requires `--wait-to-catch-up` to work properly, as otherwise it won't have the latest delayed messages to begin with).
- Confirms that delayed messages still exist before sequencing them, protecting against the inbox reader missing a reorg
- Publish batches when they fill up, not just on the hour

### CODE REVIEW

File: packages/arb-rpc-node/batcher/sequencerBatcher.go [hunk 1]
```diff
@@ -508,10 +509,27 @@ func (b *SequencerBatcher) deliverDelayedMessages(chainTime inbox.ChainTime) (bo
...
+	_, isSimulatedBackend := b.client.(*ethutils.SimulatedEthClient)
+	var getDelayedAccTarget *big.Int
+	if isSimulatedBackend {
+		// The simulated backend doesn't support querying against old blocks
+		getDelayedAccTarget = chainTime.BlockNum.AsInt()
+	} else {
+		// Confirm that the message wasn't reorganized forwards to a further reorganizable block
+		getDelayedAccTarget = lastConfirmedL1Block
+	}
+	l1DelayedAcc, err := b.inboxReader.GetDelayedAccumulator(ctx, lastDelayedSeqNum, getDelayedAccTarget)
```
Comment (hkalodner) [inline packages/arb-rpc-node/batcher/sequencerBatcher.go:540]: We should discuss whether adding an archive node requirement here is the only way to check this. In particular, could we use an event query instead?

File: packages/arb-rpc-node/batcher/sequencerBatcher.go [hunk 2]
```diff
@@ -739,6 +751,15 @@ func (b *SequencerBatcher) publishBatch(ctx context.Context, dontPublishBlockNum
 		return false, err
 	}
 
+	if publishingAllBatchItems {
+		// Reset the pending gas estimate to gasCostBase,
+		// plus whatever was sequenced while we were building this batch.
+		atomic.AddInt64(&b.pendingBatchGasEstimateAtomic, int64(gasCostBase)-origEstimate)
+	} else {
+		// Since we didn't publish everything, only subtract gas for what we did publish.
+		atomic.AddInt64(&b.pendingBatchGasEstimateAtomic, -int64(estimatedGasCost))
```
Comment (hkalodner) [inline packages/arb-rpc-node/batcher/sequencerBatcher.go:760]: In this case, doesn't the base gas cost of the next batch still need to be incorporated into the total?

**Review by hkalodner — CHANGES_REQUESTED** [review]

File: packages/arb-rpc-node/batcher/sequencerBatcher.go (hunk 1 above)
Comment (PlasmaPower) [reply packages/arb-rpc-node/batcher/sequencerBatcher.go:540]: I believed that geth would keep the last 255 blocks in memory or so. This only queries from 12 blocks ago. Is that an issue? I can update it to use an event query if so.

File: packages/arb-rpc-node/batcher/sequencerBatcher.go (hunk 2 above)
Comment (PlasmaPower) [reply packages/arb-rpc-node/batcher/sequencerBatcher.go:760]: Ah, good call

File: packages/arb-rpc-node/batcher/sequencerBatcher.go (hunk 1 above)
Comment (hkalodner) [reply packages/arb-rpc-node/batcher/sequencerBatcher.go:540]: Ah good point, I hadn't considered that the backwards looking would be tightly bounded. We'll definitely want to verify that our geth node, alchemy, and infura can handle this, but it seems safe

Comment (hkalodner) [reply packages/arb-rpc-node/batcher/sequencerBatcher.go:540]: Alchemy, infura, and default geth all look good

**Review by hkalodner — DISMISSED** [review]
LGTM

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Review by hkalodner — DISMISSED** [review]
LGTM

**Review by joshuacolvin0 — DISMISSED** [review]
LGTM

**Review by hkalodner — DISMISSED** [review]
LGTM

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Pull request merged**